src/hooks/useENSRegistration.ts

```diff
@@ -1,130 +1,88 @@
-      return {
-        code: ensValidation.code,
-        hint: ensValidation.hint,
-        valid: false,
-      };
-    }
+  const getRegistrationValues = useCallback(
+    async (_, { name, duration, nativeCurrency }) => {
+      const ensValidation = validateENS(`${name}.eth`, {
+        includeSubdomains: false,
+      });
+      if (!ensValidation.valid) {
+        return {
+          code: ensValidation.code,
+          hint: ensValidation.hint,
+          valid: false,
+        };
+      }
 
-    const isAvailable = await getAvailable(name);
-    if (isAvailable) {
-      const rentPrice = await getRentPrice(name, duration * secsInYear);
-      const nativeAssetPrice = ethereumUtils.getPriceOfNativeAssetForNetwork(
-        Network.mainnet
-      );
-      const formattedRentPrice = formatRentPrice(
-        rentPrice,
-        duration,
-        nativeCurrency,
-        nativeAssetPrice
-      );
+      const isAvailable = await getAvailable(name);
+      if (isAvailable) {
+        const rentPrice = await getRentPrice(name, duration * secsInYear);
+        const nativeAssetPrice = ethereumUtils.getPriceOfNativeAssetForNetwork(
+          Network.mainnet
+        );
+        const formattedRentPrice = formatRentPrice(
+          rentPrice,
+          duration,
+          nativeCurrency,
+          nativeAssetPrice
+        );
 
-      return {
-        available: isAvailable,
-        rentPrice: formattedRentPrice,
-        valid: true,
-      };
-    } else {
-      // we need the expiration and registration date when is not available
-      const registrationDate = await fetchRegistrationDate(name + '.eth');
-      const nameExpires = await getNameExpires(name);
-      const formattedRegistrarionDate = formatTime(registrationDate);
-      const formattedExpirationDate = formatTime(nameExpires);
+        return {
+          available: isAvailable,
+          rentPrice: formattedRentPrice,
+          valid: true,
+        };
+      } else {
+        // we need the expiration and registration date when is not available
+        const registrationDate = await fetchRegistrationDate(name + '.eth');
+        const nameExpires = await getNameExpires(name);
+        const formattedRegistrarionDate = formatTime(registrationDate);
+        const formattedExpirationDate = formatTime(nameExpires);
 
-      return {
-        available: isAvailable,
-        expirationDate: formattedExpirationDate,
-        registrationDate: formattedRegistrarionDate,
-        valid: true,
-      };
-    }
-  }, [name, duration, nativeCurrency]);
+        return {
+          available: isAvailable,
+          expirationDate: formattedExpirationDate,
+          registrationDate: formattedRegistrarionDate,
+          valid: true,
+        };
+      }
+    },
+    []
+  );
 
   const { data, status } = useQuery(
-    isValidLength && ['getRegistrationValues', name],
+    isValidLength && [
+      'getRegistrationValues',
+      { duration, name, nativeCurrency },
```

Comment:
Looks like we weren't using RQ properly, we have to pass dependencies as an input to the async function, rather than dependencies to the `useCallback`